LiveChat/components/Chat.js

```diff
@@ -11,118 +11,24 @@ const totalSize = num => (Math.sqrt((height * height) + (width * width)) * num)
 export default class Chat extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {
-      messages: [],
-      onlineStatus: false,
-      typingText: null,
-      users: {
-        system: {
-          name: 'system',
-          _id: 'system',
-        },
-      },
-    };
-    GLOBAL.visitorSDK.on('new_message', this.handleNewMessage.bind(this));
-    GLOBAL.visitorSDK.on('agent_changed', this.handleAgentChanged.bind(this));
-    GLOBAL.visitorSDK.on('status_changed', this.handleStateChange.bind(this));
-    GLOBAL.visitorSDK.on('typing_indicator', this.handleTypingIndicator.bind(this));
-    GLOBAL.visitorSDK.on('chat_ended', this.handleChatEnded.bind(this));
-    GLOBAL.visitorSDK.on('visitor_data', this.hendleVisitorData.bind(this));
 
-    this.handleInputTextChange = this.handleInputTextChange.bind(this);
-    this.handleSend = this.handleSend.bind(this);
     this.renderFooter = this.renderFooter.bind(this);
   }
 
-  getVisitor = () => {
-    const visitorId = Object.keys(this.state.users).find(userId => this.state.users[userId].type === 'visitor');
-    return this.state.users[visitorId];
-  };
-
-  handleNewMessage = (newMessage) => {
-    this.addMessage(newMessage);
-  };
-
-  addMessage = (message) => {
-    this.setState({
-      messages: [{
-        text: message.text,
-        _id: message.id,
-        createdAt: message.timestamp,
-        user: this.state.users[message.authorId],
-      }, ...this.state.messages],
-    });
-  };
-
-  handleAgentChanged = (newAgent) => {
-    this.addUser(newAgent, 'agent');
-  };
-
-  hendleVisitorData = (visitorData) => {
-    this.addUser(visitorData, 'visitor');
-  };
-
-  handleStateChange = (statusData) => {
-    this.setState({
-      onlineStatus: statusData.status === 'online',
-    });
-  };
-
-  handleInputTextChange = (text) => {
-    GLOBAL.visitorSDK.setSneakPeek({ text });
-  };
-
-  handleChatEnded = () => {
-    this.setState({
-      messages: [{
-        text: 'Chat is closed',
-        _id: String(Math.random()),
-        createdAt: Date.now(),
-        user: {
-          _id: 'system',
-        },
-      }, ...this.state.messages],
-    });
-  };
-
   handleSend = (messages) => {
```

Comment:
As you are just using inside this function the first element from the `messages` you could just deconstruct the input array on the function argument level.
```suggestion
  handleSend = ([firstMessage]) => {
```